bee-api/bee-rest-api/src/endpoints/routes/api/v1/messages_find.rs

```diff
@@ -40,35 +34,35 @@ pub(crate) fn filter<B: StorageBackend>(
                 ))),
             }
         }))
-        .and(with_storage(storage))
-        .and_then(|index, storage| async move { messages_find(index, storage) })
+        .and(with_args(args))
+        .and_then(|index, args| async move { messages_find(index, args) })
         .boxed()
 }
 
 pub(crate) fn messages_find<B: StorageBackend>(
     index: String,
-    storage: ResourceHandle<B>,
+    args: Arc<ApiArgsFullNode<B>>,
 ) -> Result<impl Reply, Rejection> {
     let index_bytes = hex::decode(index.clone())
         .map_err(|_| reject::custom(CustomRejection::BadRequest("Invalid index".to_owned())))?;
     let hashed_index = IndexationPayload::new(&index_bytes, &[]).unwrap().padded_index();
 
-    let mut fetched = match Fetch::<PaddedIndex, Vec<MessageId>>::fetch(&*storage, &hashed_index).map_err(|_| {
-        reject::custom(CustomRejection::ServiceUnavailable(
-            "can not fetch from storage".to_string(),
-        ))
-    })? {
-        Some(ids) => ids,
-        None => vec![],
-    };
+    let mut fetched =
+        match Fetch::<PaddedIndex, Vec<MessageId>>::fetch(&*args.storage, &hashed_index).map_err(|_| {
+            reject::custom(CustomRejection::ServiceUnavailable(
+                "can not fetch from storage".to_string(),
+            ))
+        })? {
+            Some(ids) => ids,
+            None => vec![],
+        };
```

Comment:
I would probably break this down into a couple statements.

Reply:
done